creator-node/src/routes/files.js

```diff
@@ -940,6 +941,54 @@ router.post(
   })
 )
 
+router.post(
+  '/segment_to_cid',
+  handleResponse(async (req, _res) => {
+    const { cid, wallet } = req.body
+    if (!cid || !wallet) {
+      return errorResponseBadRequest(`cid or wallet is missing`)
+    }
+
+    const uuid = await models.CNodeUser.findOne({
+      attributes: ['cnodeUserUUID'],
+      raw: true,
+      where: {
+        walletPublicKey: wallet
+      }
+    })
+
+    if (!uuid) {
+      return errorResponseNotFound(
+        `could not find the user uuid with the wallet provided`
+      )
+    }
+
+    const queryResults = await sequelize.query(
+      `
+      SELECT * 
+      FROM "Files" 
+      WHERE "sourceFile" in (
+        SELECT "sourceFile" 
+        FROM "Files" 
+        WHERE "multihash" = :trackSegmentCid
+      )
+      AND "type" = 'copy320'
+      AND "cnodeUserUUID" = :cnodeUserUUID
+    `,
+      {
+        replacements: {
+          trackSegmentCid: cid,
+          cnodeUserUUID: uuid
+        }
+      }
+    )
+
+    const copy320Cid = queryResults
+
+    return successResponse(copy320Cid)
```

Comment:
main concern is edge case where multiple trackIDs contain same trackSegment. in that case a copy320 could map to multiple trackIDs

i think in this case it would be good to return some more info, maybe array of tuples of type (copy320cid, array of segmentCIDs)

this way we can compare against discovery track segments array and hopefully associate correct one

Reply:
This isn't a batch route so a mapping like that could be done on the client side

Reply:
nah this info is only avail in the CN database, i'm not talking about the batch mapping. given a single segmentCID, that segment could map to multiple tracks, so we should return an array in response, and in order to distinguish the tracks, each track should be represented by a tuple of type (segmentCID[], copy320CID)

Reply:
bumping above comment

Reply:
bumping above comment

Reply:
I think I understand now. I'm still hoping the other fields I'm pulling will provide enough context to differentiate tracks if there are more than one

Reply:
i think the query will return multiple rows if there are multiple matches right? that means we'll get all the tracks that have the given segment cid. we can see which of those tracks don't have a track cid mapping in DN and populate those.